### DISCUSSION TITLE
Migrating from Bull to Bull mq

### DISCUSSION THREAD

**Discussion by RuslanBurgun** [taskforcesh/bullmq#2482] [Q&A] [answered]
Hello guys, If  migrate our project from Bull to BullMQ, will the existing jobs be executed successfully in BullMQ (which were created by bull)? Thanks.

**Comment by manast** [comment, marked as answer]
Bull and BullMQ have diverged too much now to actually give any guarantee of backwards compatibility. So the safest would be to use new queues for BullMQ and deprecate the old ones.

**Reply by RuslanBurgun** [reply]
Thank you for a quick response.

**Reply by ArkaSorice98** [reply]
Hi @manast do you mean to say that the existing queues in redis should be removed if we migrate from bull queue to bullmq? Can you please elaborate once on this.

**Reply by roggervalf** [reply]
Hi @ArkaSorice98, you can create new queues using bullmq by using different queueNames or passing a custom prefix. You can have a period where bull and bullmq queues are running at the same time where you should wait for bull queues to be drained, take in count that producers should add jobs in bullmq queues in this period. When all queues from bull are drained, you can deprecated them.

**Reply by ArkaSorice98** [reply]
Thanks for quick response
